### ISSUE TITLE
Challenges understanding error in NS3 lorawan module

### ISSUE THREAD

**Issue by SonileKM** [signetlabdei/lorawan#170]
## Actual Behavior
This is the error we are getting: The error shown in the screen shot attached and we can' seem to figure out what the real issue is and to get around it.

## Steps to Reproduce the Problem  
  1.
  2.
  3.

## Specifications

  - ns-3 version: 3.40
  - lorawan module version: v0.3.1
  - Platform: Ubuntu linux
  - Compiler: gcc
![Screenshot from 2024-07-23 14-28-26](https://github.com/user-attachments/assets/9002068c-1295-4868-93d6-415bcecb4eef)

**Comment by nondetalle** [comment]
Hello, release v0.3.1 of the lorawan module depends on ns-3.41. You are using ns-3.40, that's what is causing the errors.

**Comment by SonileKM** [comment]
Thanks for your response Alessandro. We actually installed ns-341 and it is
giving us an error that we are failing to resolve. The compiler is only
recognising header files in the lorawan module but not those that are in
the folder for the source code we are trying to run. Here is a snapshot of
the output we are getting if we run it from the ns-3.41 directory or from
the ns-3-dev directory.

**Comment by nondetalle** [comment]
> Here is a snapshot of the output we are getting if we run it from the ns-3.41 directory or from the ns-3-dev directory.

Hi @SonileKM, I cannot find the snapshot you mention.

**Comment by SonileKM** [comment]
Please see the attachment for the screenshot. Please advise if you are able
to view it.

**Comment by nondetalle** [comment]
Hi. Still not able to view it. If you need to attach an image, please reply using the interface on the GitHub website.

**Comment by SonileKM** [comment]
Following your advice, here it the attachment. 
![Screenshot from ns-3-dev](https://github.com/user-attachments/assets/6e937ce5-03b2-4534-b40e-35fbe459d2c7)

**Comment by nondetalle** [comment]
> The compiler is only recognising header files in the lorawan module but not those that are in the folder for the source code we are trying to run.

Hi @SonileKM, where did you place `gw-trailer.h` and the other files you want to include in `rs-example.cc`? This is a bit out of scope for us to help you, given that the files of this repo are not the source of your errors. 

If you need to run a simulation with additional libraries, maybe refer to the multi-file examples provided by ns-3 in the [scratch directory](https://gitlab.com/nsnam/ns-3-dev/-/tree/ns-3.41/scratch). If you want to create a new ns-3 module with the files you need to import, checkout the ns-3 manual on [how to create a new module](https://www.nsnam.org/docs/manual/singlehtml/index.html#adding-a-new-module-to-ns3) instead. Finally, if you want to  integrate your additional files in a copy of this repository, I suggest you start studying both [the model](https://signetlabdei.github.io/lorawan/models/build/html/lorawan.html) and the code directly to understand where you could fit the additional functionality.

**Comment by SonileKM** [comment]
Thank you so much Alessandro. I will definitely follow through with the
suggestions you have given.

Regards,
Sonile

**Comment by SonileKM** [comment]
I place gw-trailer and other header files in a model folder which is inside
the project folder. The rs-example code is also in the example folder
inside the project folder.

**Comment by SonileKM** [comment]
I forgot something. I copied the rs-example code on the scratch folder of
ns-341, so i was running this code in ns3.

**Comment by nondetalle** [comment]
Ok, so if you are putting your library files (`gw-trailer.h`, `gw-trailer.cc`, etc.) in the [model](/signetlabdei/lorawan/blob/develop/model) directory of this repo, you should also add them in [CMakeLists.txt](/signetlabdei/lorawan/blob/develop/CMakeLists.txt) to allow the ns-3 build system to pick them up at compilation time. Make sure to put everything inside the `ns3` namespace, otherwise include directories as `#include "ns3/gw-trailer.h"` won't work.

If that does not produce compilation errors, try running `rs-example.cc` as you did before. It's ok to leave it in the `scratch` folder (the main difference is that in `scratch` files are automatically added to the build system without having to specify them in a `CMakeLists.txt` file, as you would need to do instead in the `examples` directory).

**Comment by SonileKM** [comment]
Thank you so much Alessandro. I will try this.

Regards,
Sonile

**Comment by SonileKM** [comment]
Dear Alessandro,
Thank you so much for this valuable advice. The error due to external
library files (gw-trailer.h, ge-trailer.cc) has been resolved. I will
definitely get back to you if i encounter other challenges.
